aleph/views/bookmarks_api.py

```diff
@@ -1,51 +1,88 @@
 import logging
 from flask import Blueprint, request
-from aleph.views.util import jsonify, require
+
+from aleph.core import db
+from aleph.search import DatabaseQueryResult
+from aleph.views.util import jsonify, require, parse_request
+from aleph.views.serializers import BookmarkSerializer
+from aleph.model import Bookmark
 
 
 log = logging.getLogger(__name__)
 blueprint = Blueprint("bookmarks_api", __name__)
 
 
+@blueprint.route("/api/2/bookmarks", methods=["GET"])
+def index():
+    """Get a list of bookmarks created by the current user
+    ---
+    get:
+      summary: Get bookmarks
+      tags: [Bookmarks]
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json
+              schema:
+                type: object
+    """
+    require(request.authz.logged_in)
+
+    query = Bookmark.query.filter_by(role_id=request.authz.id)
+    result = DatabaseQueryResult(request, query)
+    return BookmarkSerializer.jsonify_result(result)
+
+
 @blueprint.route("/api/2/bookmarks", methods=["POST"])
 def create():
-    """Creates a new bookmark for the user. This is currently a stub endpoint,
-    as the current version of the bookmarks features stores bookmarks client side.
+    """Bookmark one or more entities.
     ---
     post:
-      summary: Create bookmark
+      summary: Create bookmarks
+      tags: [Bookmarks]
       responses:
         '200':
+          description: OK
           content:
             application/json:
               schema:
                 type: object
-          description: OK
-      tags:
-        - Bookmarks
     """
-    require(request.authz.logged_in)
-    log.info("User [%s]: Created bookmark", request.authz.role.id)
-    return jsonify({}, 200)
+    require(request.authz.session_write)
+    data = parse_request("BookmarkBulkCreate")
+
+    bookmarks = [
+        Bookmark(entity_id=bookmark.get("entity_id"), role_id=request.authz.id)
+        for bookmark in data
+    ]
+    db.session.add_all(bookmarks)
```

Comment:
Could be implemented using SQLAlchemy’s [`bulk_save_objects`](https://docs.sqlalchemy.org/en/14/orm/session_api.html#sqlalchemy.orm.Session.bulk_save_objects). This feature is deprecated though, but SQLAlchemy 2 has an equivalent replacement (I think -- see docs).

Reply:
Apart from that, I’m not sure whether bulk creation should be handled by the default `POST /bookmarks` endpoint or whether it should be a separate endpoint (that we could potentially remove again some time after this feature has been released). Might lead to a cleaner API design and simplify semantics?